internal/terraform/exec/exec.go

```diff
@@ -210,65 +124,30 @@ func (e *Executor) FormatterForVersion(v string) (Formatter, error) {
 	}
 
 	if ver.GreaterThanOrEqual(fmtCapableVersion) {
-		return e.Format, nil
+		return f, nil
 	}
 
 	return nil, fmt.Errorf("no formatter available for %s", v)
 }
 
 func (e *Executor) Format(ctx context.Context, input []byte) ([]byte, error) {
-	cmd, err := e.cmd(ctx, "fmt", "-")
-	if err != nil {
-		return nil, err
-	}
+	ctx, cancel := contextWithTimeout(ctx, e.timeout)
+	defer cancel()
 
-	stdin, err := cmd.Cmd.StdinPipe()
-	if err != nil {
-		return nil, err
-	}
-
-	err = cmd.Cmd.Start()
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = writeAndClose(stdin, input)
-	if err != nil {
-		return nil, err
-	}
-
-	out, err := e.waitCmd(cmd)
-	if err != nil {
-		return nil, fmt.Errorf("failed to format: %w", err)
-	}
-
-	return out, nil
-}
-
-func writeAndClose(w io.WriteCloser, input []byte) (int, error) {
-	defer w.Close()
-
-	n, err := w.Write(input)
-	if err != nil {
-		return n, err
-	}
-
-	return n, nil
+	formatted, err := tfexec.FormatString(ctx, e.execPath, string(input))
+	return []byte(formatted), err
 }
 
 func (e *Executor) Version(ctx context.Context) (string, error) {
-	out, err := e.run(ctx, "version")
+	ctx, cancel := contextWithTimeout(ctx, e.timeout)
+	defer cancel()
+
+	v, _, err := e.tfExec().Version(ctx, true)
 	if err != nil {
-		return "", fmt.Errorf("failed to get version: %w", err)
-	}
-	outString := string(out)
-	lines := strings.Split(outString, "\n")
-	if len(lines) < 1 {
-		return "", fmt.Errorf("unexpected version output: %q", outString)
+		return "", err
 	}
-	version := strings.TrimPrefix(lines[0], "Terraform v")
-
-	return version, nil
+	// TODO: consider refactoring codebase to work directly with go-version.Version
```

Comment:
I noticed a fair bit of back and forth between `go-version.Version` and the stringified representation. I think given I saw `go-version` types bled onto a few structs throughout the codebade, and given `terraform-exec` works with it directly it may be worth changing the language server codebase to work directly in that type, in a follow up PR.